packages/babel-types/src/clone/cloneNode.js

```diff
@@ -0,0 +1,35 @@
+import { NODE_FIELDS } from "../definitions";
+
+const has = Function.call.bind(Object.prototype.hasOwnProperty);
+const isNode = obj => obj && has(NODE_FIELDS, obj.type);
+
+export default function cloneNode<T: Object>(node: T): T {
```

Comment:
I'd say it would be better to extend `BabelNode` instead...... but actually reading the `BabelNode` definition and how it's used makes it a bit doubtful.

For example, in TypeScript, almost every valid node (written as an object literal) you gave it would get caught in excess property checks for weak types. It'd probably be better to make `BabelNode` be a generic, where the generic argument is the node type literal string.

But I digress and is outside the scope of this change.